File: 2021W06/fienestar/boj_1931.cpp
/*
    회의실 배정
    https://www.acmicpc.net/problem/1931
*/

#include <bits/stdc++.h>

using namespace std;
using pii = pair<int,int>;

int main()
{
    size_t N;
    cin >> N;

    vector<pii> v(N);
    for(auto& c:v)
        cin >> c.first >> c.second;

    sort(v.begin(),v.end(),[](pii a,pii b){
        if(a.second == b.second)
            return less<int>()(a.first,b.first);
        return less<int>()(a.second,b.second);
    });

    int last = 0;
    size_t cnt = 0;
    for(auto& c:v)
        if(c.first >= last){
            ++cnt;
            last = c.second;
        }
    cout << cnt;
}

File: 2021W29/B0neh3ad/boj_1931.py
'''
    회의실 배정
    https://www.acmicpc.net/problem/1931
'''
from operator import itemgetter

N = int(input())
sch = []
for i in [0]*N:
    sch.append(list(map(int, input().split())))

sch.sort(key = itemgetter(1, 0))

curr_end = sch[0][1]
cnt = 1
for con in sch[1:]:
    if curr_end <= con[0]:
        curr_end = con[1]
        cnt += 1

print(cnt)